libraries/chain/include/eosio/chain/hotstuff/finalizer.hpp

```diff
@@ -68,22 +70,38 @@ namespace eosio::chain {
       using fsi_t   = finalizer_safety_information;
       using fsi_map = std::map<bls_public_key, fsi_t>;
 
+   private:
       const block_timestamp_type        t_startup;             // nodeos startup time, used for default safety_information
       const std::filesystem::path       persist_file_path;     // where we save the safety data
+      mutable std::mutex                mtx;
       mutable fc::datastream<fc::cfile> persist_file;          // we want to keep the file open for speed
-      std::map<bls_public_key, finalizer>  finalizers;         // the active finalizers for this node
+      std::map<bls_public_key, finalizer>  finalizers;         // the active finalizers for this node, loaded at startup, not mutated afterwards
       fsi_map                           inactive_safety_info;  // loaded at startup, not mutated afterwards
       fsi_t                             default_fsi = fsi_t::unset_fsi(); // default provided at leap startup
       mutable bool                      inactive_safety_info_written{false};
 
-      template<class F>
+   public:
+      my_finalizers_t(block_timestamp_type startup_time, const std::filesystem::path& persist_file_path)
+         : t_startup(startup_time)
+         , persist_file_path(persist_file_path)
+      {}
```

Comment:
Why add this constructor which prevents aggregate initioalization?

Reply:
Because the attributes are now private. They are prviate so their access can be protected by a mutex.

Reply:
Do they need to be protected by a mutex?  I believe they are set at startup time and never change.

Reply:
Those two do not, but the other attributes do need the protection.

Reply:
got it, thanks.